### PR TITLE
Add Multi-Lingual LibriSpeech

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Add https://www.openslr.org/94/

### CODE REVIEW

File: datasets/multilingual_librispeech/multilingual_librispeech.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    """Multilingual Librispeech dataset."""
+
+    BUILDER_CONFIGS = [
+        MultilingualLibrispeechConfig(name="german", description="German LibriSpeech dataset"),
+        MultilingualLibrispeechConfig(name="dutch", description="Dutch LibriSpeech dataset"),
+        MultilingualLibrispeechConfig(name="french", description="French LibriSpeech dataset"),
+        MultilingualLibrispeechConfig(name="spanish", description="Spanish LibriSpeech dataset"),
+        MultilingualLibrispeechConfig(name="italian", description="Italian LibriSpeech dataset"),
+        MultilingualLibrispeechConfig(name="portuguese", description="Portuguese LibriSpeech dataset"),
+        MultilingualLibrispeechConfig(name="polish", description="Polish dataset"),
```
Comment (albertvillanova) [inline datasets/multilingual_librispeech/multilingual_librispeech.py:69]:
```suggestion
        MultilingualLibrispeechConfig(name="polish", description="Polish LibriSpeech dataset"),
```

File: datasets/multilingual_librispeech/multilingual_librispeech.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                }
+            ),
+            supervised_keys=("file", "text"),
+            homepage=_URL,
+            citation=_CITATION,
+            task_templates=[AutomaticSpeechRecognition(audio_file_path_column="file", transcription_column="text")],
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download_and_extract(self.config.data_dir)
```
Comment (albertvillanova) [inline datasets/multilingual_librispeech/multilingual_librispeech.py:92]: Just we should not forget that this should be refactored to use `iter_archive` once merged:
- #3110

File: datasets/multilingual_librispeech/multilingual_librispeech.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            all_ids = set(all_ids)
+
+        with open(transcript_path, "r", encoding="utf-8") as f:
+            for line in f:
+                line = line.strip()
+                id_, transcript = line.split("\t")
+
+                if all_ids is not None and id_ not in all_ids:
+                    # this only holds true for train.9h and train.1h
+                    continue
```
Comment (albertvillanova) [inline datasets/multilingual_librispeech/multilingual_librispeech.py:141]: Therefore "train" split also contains ""train.9h" and "train.1h"?

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Thanks for the addition of this great dataset, @patrickvonplaten.

Below some questions/comments and nit requested changes.

- We do not add the suffix "_asr" to the dataset name (as was done with `librispeech_asr`)? Note that there is also "_lm" data.

File: datasets/multilingual_librispeech/multilingual_librispeech.py (hunk 3 above)
Comment (patrickvonplaten) [reply datasets/multilingual_librispeech/multilingual_librispeech.py:141]: yes

File: datasets/multilingual_librispeech/multilingual_librispeech.py (hunk 2 above)
Comment (patrickvonplaten) [reply datasets/multilingual_librispeech/multilingual_librispeech.py:92]: ok, but this is not relevant for this PR no?

File: datasets/multilingual_librispeech/multilingual_librispeech.py (hunk 3 above)
Comment (patrickvonplaten) [reply datasets/multilingual_librispeech/multilingual_librispeech.py:141]: just like train.100 and train.360 for datasets

**Review by albertvillanova — APPROVED** [review]

**Pull request merged**